### ISSUE TITLE
Flash RockChip boards

### ISSUE THREAD

**Issue by samgutentag** [NebraLtd/helium-linxdot-rkcm3#2]
Any tips or tricks to flash this firmware to the Linxdot Rockchip boards? Have a few I would love to get up and running and happy to help test and document the process!

**Comment by shawaj** [comment]
the image is not quite ready yet...it will be a week or so longer to get it finished likely

will post back here when it is ready

**Comment by QuentinDowsers** [comment]
https://wiki.radxa.com/Rock3/installusb-install-radxa-cm3-io
Tips: On linux: $ rkdeveloptool rd 3 // "3" to start your equipment in maskrom

I have successfully flashed mine but ECC and serial are not detected...
My hotspot was onboarder and asserted previously
![Capture d'écran 2023-11-10 103115](https://github.com/NebraLtd/helium-linxdot-rk3566/assets/130649524/dc83afe4-3a1a-4e52-9e5d-e0dc93d8eb48)
![Capture d'écran 2023-11-10 103137](https://github.com/NebraLtd/helium-linxdot-rk3566/assets/130649524/edbb49c8-9d12-46ce-bfac-ddc3db1c4625)

**Comment by butterwecksolutions** [comment]
@QuentinDowsers How did you flash it? I tried it with a own img from balena and the img from the latest release but the linuxdot is not booting with it. I flashed with the windows RK Dev Tool.

**Comment by QuentinDowsers** [comment]
@christian1980nrw I have flashed it on linux: you need to have also the image of the bootloader for the RK chip.

The most important step: while the Hotspot is powered OFF but still connected to ethernet and your PC via USB, you need to HOLD down the BT pair button for 5 seconds and plug in the power cable. Then release the BT pair button after holding down for 5 more seconds after powering ON your Hotspot. You need to make sure the BT pair button is held down for the entirety of this process - this will enable the flashing mode

**Comment by QuentinDowsers** [comment]
[Linxdot-RK3566.pdf](https://github.com/NebraLtd/helium-linxdot-rk3566/files/14232455/Linxdot-RK3566.pdf)
Also, you can follow the procedure of Crankk OS (https://crankk.io/guides/) with the Nebra Open Fleet OS instead.

**Comment by shawaj** [comment]
@QuentinDowsers is your unit still on the firmware?

If so, with your permission I can try to remote into it and work out the issue with the ECC chip

**Comment by QuentinDowsers** [comment]
Hello @shawaj ,

Yes still running on Nebra OS and connected.
Of course, you have my permission to work on it. It would be my pleasure.

I might have an idea from https://github.com/helium/ecc508 on I2C bus...

**Comment by QuentinDowsers** [comment]
> @QuentinDowsers is your unit still on the firmware?
> 
> If so, with your permission I can try to remote into it and work out the issue with the ECC chip

Hello @shawaj ,
Any news ? Should I send you the connection details in private (because MAC address and hostname are no more the same) ?

**Comment by shawaj** [comment]
@QuentinDowsers if you could that would be great. You can email me (check my profile for email)

**Comment by anahtarr** [comment]
Hell

> @christian1980nrw I have flashed it on linux: you need to have also the image of the bootloader for the RK chip.
> 
> The most important step: while the Hotspot is powered OFF but still connected to ethernet and your PC via USB, you need to HOLD down the BT pair button for 5 seconds and plug in the power cable. Then release the BT pair button after holding down for 5 more seconds after powering ON your Hotspot. You need to make sure the BT pair button is held down for the entirety of this process - this will enable the flashing mode

Hi, I made a mistake and install armbian Orangi pi3b image to linxdot ld1001 hotspot. Now even if I press bluetooth button, I can not see any change (maskmode or found one devices) in  Rkdevtool. What can I do now? How can I get the maskmode ? 
Here someone had the same problem: https://forum.armbian.com/topic/34455-customboard-with-rk3566-flashed-with-orangepi3b-armbian-minnimal-image-no-more-maskmode-in-rkdevtool/

**Comment by lexiz52** [comment]
@QuentinDowsers : Do you have still the issue that the ECC is not detected! Or did you solve the issue? I have same on my LinxDot.

**Comment by QuentinDowsers** [comment]
> @QuentinDowsers : Do you have still the issue that the ECC is not detected! Or did you solve the issue? I have same on my LinxDot.

Yes, I still have this issue on my linxdot , @shawaj didn't reply to mail...

**Comment by shawaj** [comment]
Just replied. I need to get access to someone's device that's running Nebra firmware to check it out

**Comment by shawaj** [comment]
@QuentinDowsers ^^

**Comment by QuentinDowsers** [comment]
yes still connected and waiting for you.

**Comment by shawaj** [comment]
@QuentinDowsers I can't see it on my end. It's on balenacloud? Or what firmware file did you use?

**Comment by QuentinDowsers** [comment]
It is still on nebra OS, last conf is:
 Firmware Version	v1.3.3 (7d24221)
 Variant	Linxdot RK3566 Hotspot
 Hardware Serial Number	Serial number not found
 Hostname	nebra-C99C0C.local
 Local IP Address	
192.168.1.152    2a01:cb08:230:6900:a9d2:c0ab:7725:87a0
 WAN IP Address	XXXXXXXXXXXXXX
 Ethernet MAC Address	96:73:FD:C9:9C:0C
 WiFi MAC Address	08:E9:F6:EE:0F:74

**Comment by shawaj** [comment]
@QuentinDowsers I need the full diagnostic JSON or the device UUID from the JSON.

**Comment by shawaj** [comment]
In the JSON it would be listed as "ID"

**Comment by QuentinDowsers** [comment]
"ID":"eab115d21e752984549b4f8f17024015"

**Comment by shawaj** [comment]
@QuentinDowsers Thanks... This device says it's offline though so I can't currently do anything with it

**Comment by QuentinDowsers** [comment]
> @QuentinDowsers Thanks... This device says it's offline though so I can't currently do anything with it

That's weird? Should I open port for you on my router?
It gets updates from nebra so it can discuss with your server...

**Comment by shawaj** [comment]
@QuentinDowsers maybe it's an issue on our side. Could you try rebooting it though just in case?

**Comment by lexiz52** [comment]
> @QuentinDowsers : Do you have still the issue that the ECC is not detected! Or did you solve the issue? I have same on my LinxDot.

Thanks for your reply!

**Comment by QuentinDowsers** [comment]
> @QuentinDowsers maybe it's an issue on our side. Could you try rebooting it though just in case?

It is rebooted now

**Comment by QuentinDowsers** [comment]
@shawaj I have opened the port 22222 for balena ssh server maybe it will be usefull for you?
Or should I try to open a new port specific to nebra?

**Comment by sukerl** [comment]
@anahtarr I have the exact problem as you had. Did you find a solution to flash the Linxdot to a working image?

**Comment by GokhanZukachi** [comment]
I have the exact problem too. I think linxdot rockchip nebra image is not working properly. Is there anyone here Using linxdot nebra image properly?

**Comment by juba86** [comment]
got the same problem

**Comment by juba86** [comment]
any updates here?

**Comment by tomdickharryeth** [comment]
@shawaj in case it helps, i am having the same problem and i'm seeing some kind of DNS resolution error in the diagnostics

"gateway_pubkey": "<_InactiveRpcError of RPC that terminated with:\n\tstatus = StatusCode.UNAVAILABLE\n\tdetails = \"DNS resolution failed for helium-miner:4467: C-ares status is not ARES_SUCCESS qtype=A name=helium-miner is_balancer=0: DNS server returned answer with no data\"\n\tdebug_error_string = \"UNKNOWN:DNS resolution failed for helium-miner:4467: C-ares status is not ARES_SUCCESS qtype=A name=helium-miner is_balancer=0: DNS server returned answer with no data {grpc_status:14, created_time:\"2023-01-03T04:11:55.00433984+00:00\"}\"\n>",
"gateway_region": "<_InactiveRpcError of RPC that terminated with:\n\tstatus = StatusCode.UNAVAILABLE\n\tdetails = \"DNS resolution failed for helium-miner:4467: C-ares status is not ARES_SUCCESS qtype=A name=helium-miner is_balancer=0: DNS server returned answer with no data\"\n\tdebug_error_string = \"UNKNOWN:DNS resolution failed for helium-miner:4467: C-ares status is not ARES_SUCCESS qtype=A name=helium-miner is_balancer=0: DNS server returned answer with no data {grpc_status:14, created_time:\"2023-01-03T04:11:55.14645384+00:00\"}\"\n>",

**Comment by frostfang** [comment]
Any news on this? wondering if anyone managed to get this working?
